internal/storage/db/internal/db.go

```diff
@@ -652,6 +654,34 @@ func (s *dbStorage) Delete(ctx context.Context, ids ...namer.ModuleID) error {
 	return nil
 }
 
+func (s *dbStorage) InspectPolicies(ctx context.Context, listParams storage.ListPolicyIDsParams) (map[string]*responsev1.InspectPoliciesResponse_Result, error) {
+	policyIDs, err := s.ListPolicyIDs(ctx, listParams)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list policies: %w", err)
+	}
+
+	if len(policyIDs) == 0 {
+		return nil, nil
+	}
+
+	policies, err := s.LoadPolicy(ctx, policyIDs...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load policies: %w", err)
+	}
+
+	results := make(map[string]*responsev1.InspectPoliciesResponse_Result)
+	for _, p := range policies {
+		actions := policy.ListActions(p.Policy)
+		if len(actions) > 0 {
+			results[p.FQN] = &responsev1.InspectPoliciesResponse_Result{
+				Actions: actions,
+			}
+		}
+	}
```

Comment:
Instead of loading all the policies into memory at once, load them in small batches. That's more efficient in terms of database usage and memory utilization.